v22.1/vault-db-secrets-tutorial.md

```diff
@@ -0,0 +1,441 @@
+---
+title: Using Hashicorp Vault's Dynamic Secrets for Enhanced Database Credential Security
```

Comment:
A few general notes, adding Steps + numbers for the H3 headers will align with the [style guide](https://github.com/cockroachdb/docs/blob/master/StyleGuide.md#tutorial). 

I think there is room for you to make the directions much more prescriptive. In this way, it will be much harder for the reader to get off track and run into errors. Although, certain small tasks might seem obvious to us, spelling them out will help in this effort to make it smooth for the reader. 

For example, telling the reading when they need to add in their own details to a command or variable, because the code block is just a placeholder would make a big difference. And breaking up the code blocks so the reader can see the separate actions they need to perform a little clearer.